packages/components/pie-modal/src/index.ts

```diff
@@ -44,20 +44,62 @@ export class PieModal extends RtlMixin(LitElement) {
                         variant="ghost-tertiary"
                         class="c-modal-closeBtn"></pie-icon-button>
                 </header>
-                <article>
-                    <div class="c-modal-content">
-                        <slot></slot>
-                    </div>
+                <article class="c-modal-content">
+                    <slot></slot>
                 </article>
             </dialog>
         `;
     }
 
+    updated () {
+        this._handleDialogLightDismiss();
+    }
+
+    /**
+     * We require this because toggling the prop `isOpen` itself won't
+     * allow the dialog to open in the correct way (with the default background),
+     * the method `showModal()` needs to be invoked.
+     *
+     * https://lit.dev/docs/components/lifecycle/#willupdate
+     *
+     * @param changedProperties
+     */
+    willUpdate (changedProperties: Map<string, unknown>) {
+        if (changedProperties.has('isOpen') && this._dialog) {
+            const dialog = this._dialog;
+            const isClosed = changedProperties.get('isOpen') === false;
+
+            if (isClosed) {
+                dialog.showModal();
+            } else {
+                dialog.close();
+            }
+        }
+    }
+
     _handleCloseDialog = () => {
-        this._dialog.close();
+        this._dialog?.close();
         this._onDialogClose();
     };
 
+    /**
+     * Dismiss modal via `_handleCloseDialog()` on backdrop click
+     * and will proceed to fire an `ON_MODAL_CLOSE_EVENT` event.
+     *
+     */
+    _handleDialogLightDismiss = () => {
+        this._dialog?.addEventListener('click', (event) => {
```

Comment:
Will this add a new click handler every time the modal updates?

Reply:
Good point.

Reply:
piggy backing this, do we need to consider removing any event listeners on the component being destroyed? I assume if the listener is on said component then perhaps not?

Reply:
I'm still a bit uncertain on how lit deals with events if i'm honest. Their documentation suggests different ways to add them and some dont actually say you need to remove it either. They only mention removing listeners via the connected / discounted hooks and there is also a section about passing null into the event to remove the listener itself. I am fairly certainwe dont need to remove them.

`No need to remove internal event listeners. You don't need to remove event listeners added on the component's own DOM—including those added declaratively in your template. Unlike external event listeners, these won't prevent the component from being garbage collected.`

Reply:
Yeah I think in this instance you're all good!

Reply:
I think there may still be an issue with adding too many event listeners but I don't mind if you want to test that post-merge

Reply:
I think they get removed but will have a play locally once merged. From what I see Lit destroys them for you.

Reply:
I would move the event listener added to the constructor as outlined in [Lit's docs](https://lit.dev/docs/components/events/#adding-event-listeners-to-the-component-or-its-shadow-root)

Reply:
Moved to constructor.